### DISCUSSION TITLE
Event only fires when created on admin ui

### DISCUSSION THREAD

**Discussion by muhammed-kilinc** [pocketbase/pocketbase#2047] [Q&A] [answered]
Hello guys I have a event which should fire on new Order but it only fires when i create the new order via admin ui 
```go
app.OnRecordAfterCreateRequest("orders").Add(func(e *core.RecordCreateEvent) error {
		println("NEW ORDER")
		message := &mailer.Message{
			From: mail.Address{
				Address: app.Settings().Meta.SenderAddress,
				Name:    app.Settings().Meta.SenderName,
			},
			To: []mail.Address{
				{
					Address: e.Record.GetString("to"),
				},
			},
			Subject: "New Order",
			HTML: "<div><p>Hi, you've recieved a new Order on gomentors.online</p></div>" +
				"<a href='http://localhost:3000/Messages/" + e.Record.GetId() + "'>Click to view your message</a>",
			//NEW_ORDER_HTML
			// bcc, cc, attachments and custom headers are also supported...
		}
		println("Sending mail to " + e.Record.GetString("to"))
		return app.NewMailClient().Send(message)
	})
```

The way I am creating the Order : 
```go
func (api *orderApi) createOrder(c echo.Context) (err error) {
	params := c.QueryParams()

	collection, err := api.app.Dao().FindCollectionByNameOrId("orders")

	record := models.NewRecord(collection)

	form := forms.NewRecordUpsert(api.app, record)
	form.LoadData(map[string]any{
		"session":  params.Get("session"),
		"hours":    params.Get("hours"),
		"message":  params.Get("message"),
		"creator":  params.Get("creator"),
		"reciever": params.Get("reciever"),
		"to":       params.Get("to"),
	})
	if err := form.Submit(); err != nil {
		return err
	} else {
		return c.Redirect(303, "http://localhost:3000/orders")
	}
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
This is the same as https://github.com/pocketbase/pocketbase/discussions/2045.

`OnRecordAfterCreateRequest` is triggered only on API request. If you want to listen to programmatically inserted record events, you can bind to the [`OnModelAfterCreate` db/Dao event hooks](https://pocketbase.io/docs/event-hooks/#db-hooks).

**Reply by ganigeorgiev** [reply]
_Side-note: to get a `Record` struct from inside a `OnModel*` hook, you have to use type assertion:_
```go
app.OnModelAfterCreate("orders").Add(func(e *core.ModelEvent) error {
	record := e.Model.(*models.Record)
        ...
})
```

**Reply by muhammed-kilinc** [reply]
am I supposed to use app.Dao or e.Dao if needed ?

**Reply by ganigeorgiev** [reply]
The `ModelEvent.Dao` refers to the `Dao` that triggered this event.

For the `OnModelBefore*` events  in case of a transaction, it'll be a transactional `Dao` and it allows to execute db statements within the running transaction.

For the `OnModelAfter*` events most of the time `e.Dao` will be the same as `app.Dao()` (_because once a transaction is committed the connection must be discarded_).

To summarize, for the after events it shouldn't matter and you can use whatever `Dao` is available.

**Reply by structwafel** [reply]
> _Side-note: to get a `Record` struct from inside a `OnModel*` hook, you have to use type assertion:_
> 
> ```go
> app.OnModelAfterCreate("orders").Add(func(e *core.ModelEvent) error {
> 	record := e.Model.(*models.Record)
>         ...
> })
> ```

Unrelated, but thankyou for this side-note. Was stuck on this for to long.
